tools/tesseratos/src/tesseratos/toolbox/plugin.cpp

```diff
@@ -0,0 +1,71 @@
+#include <imgui.h>
+
+#include <cubos/engine/imgui/plugin.hpp>
+
+#include <tesseratos/toolbox/plugin.hpp>
+
+using cubos::core::ecs::EventWriter;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using cubos::core::ecs::Write;
+using cubos::engine::Cubos;
+
+using namespace tesseratos;
+
+bool Toolbox::isOpen(std::string toolName)
+{
+    if (mToolsMap.contains(toolName))
+    {
+        return mToolsMap[toolName];
+    }
+    mToolsMap[toolName] = false;
+    return false;
+}
+
+void Toolbox::open(std::string toolName)
+{
+    mToolsMap[toolName] = true;
+}
+
+void Toolbox::close(std::string toolName)
+{
+    mToolsMap[toolName] = false;
+}
+
+void Toolbox::doClick(std::string toolName)
+{
+    mToolsMap[toolName] = !mToolsMap[toolName];
```

Comment:
Should we have a some kind of check here?
For tools that keep an internal state it doesn't make sense to me to be able to just close them while that state is altered.
For example, the palette editor. If there's an unsaved edit, does it make sense to have those unsaved changes hidden in a closed tool?

Reply:
This is indeed a problem... we should discuss it in the meeting. One solution is to just treat this as 'hiding' the tool and not closing anything.

Reply:
That is very error prone imo. You close the tool with unsaved changes, spend two hours working on other stuff, based on the unsaved changes you have, and then close the program without ever saving because you forget there was dirty stuff in the background